src/Discovery/DiscoveryActionBar.tsx

```diff
@@ -483,21 +512,24 @@ const DiscoveryActionBar = (props: Props) => {
           arrowPointAtCenter
           content={(
             <React.Fragment>
-          Directly download data (up to 250Mb) from selected studies
+              {healICPSRLoginNeeded
+                ? 'This dataset is only accessible to users who have authenticated via InCommons. Please log in using the InCommons option.'
+                : 'Directly download data (up to 250Mb) from selected studies'}
             </React.Fragment>
           )}
         >
           <Button
             onClick={
               async () => {
-                if (props.user.username) {
+                if (props.user.username && !healICPSRLoginNeeded) {
```

Comment:
can we make the same restriction applied to all 3 buttons in this component, in other words, can we apply the same logic also to the manifest download button and the export to workspace button?

Reply:
yes, i will do that